### PR TITLE
batches: also show the missing credential alert on batch changes

### LINKED ISSUE #40227: batches: ensure credential banner is shown when publishing via a bulk operation
[Slack thread for context.](https://sourcegraph.slack.com/archives/C03CSAER9LK/p1660161193958949?thread_ts=1660161193958949&cid=C03CSAER9LK)

The banner we should to say "you don't have credentials for this code host" isn't showing up in this case, where a changeset was published via a bulk operation instead of directly. We should check that this behaves as expected.

### PR DESCRIPTION
**Pull request by LawnGnome**
My investigation of #40227 led me to realise that we show the lovely missing credential alert banner on the preview pages when the user is about to apply a batch change, but not on the actual batch change itself. This made sense before the primary flow to publish changesets was driven by the spec and/or the preview page, but this may make less sense in a world where there's a publish bulk operation.

This PR adds the same banner to batch changes that have one or more unpublishable changesets. I'm not _totally_ sure that we want this, but I suspect we do.

I'm OK holding this for 4.1, so I haven't added the 4.0 label for now, but I'd like input on that as well.

Fixes #40227.

## Test plan

Tested the two preview page permutations (SSBC and non-SSBC), along with the batch change details page with and without credentials. Also added storybook coverage for the new case.

## App preview:

- [Web](https://sg-web-aharvey-add-cred-banner-to-details.onrender.com)
- [Storybook](https://5f0f381c0e50750022dc6bf7-ozjmhbvhfz.chromatic.com)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.

### CODE REVIEW

File: client/web/src/enterprise/batches/MissingCredentialsAlert.tsx [hunk 1]
```diff
@@ -36,7 +48,7 @@ export const MissingCredentialsAlert: React.FunctionComponent<
                 <Link to={`${authenticatedUser.url}/settings/batch-changes`} target="_blank" rel="noopener">
                     batch changes user settings
                 </Link>{' '}
-                to apply this spec.
+                to publish changesets from this batch change.
```
Comment (eseliger) [inline client/web/src/enterprise/batches/MissingCredentialsAlert.tsx:51]: nice wording tweak!

**Review by eseliger — APPROVED** [review]
In absence of design resources in the next few weeks, I think I'm ok with that change for now. We might want to rethink it a little as we got quite a few banners that could show simultaneously, but I don't think that should block this improvement.

**Review by courier-new — APPROVED** [review]
Yeah works for me for now, too. We already have too many alerts on the BC details page, 1 more won't hurt it. 😛

File: client/web/src/enterprise/batches/detail/backend.ts [hunk 2]
```diff
@@ -171,6 +172,9 @@ const batchChangeFragment = gql`
                 }
                 totalCount
             }
+            viewerBatchChangesCodeHosts(onlyWithoutCredential: true) {
```
[the remark this answers is not in the document]
Comment (LawnGnome) [reply client/web/src/enterprise/batches/detail/backend.ts:175]: I personally think the `onlyWithoutCredential` parameter name on the field is clear enough here, but if you feel strongly about it, I'm happy to take this in a follow up.

[the remark this answers is not in the document]
Comment (eseliger) [reply client/web/src/enterprise/batches/detail/backend.ts:175]: same

**Pull request merged**